commit 97abfa4e78f3c467de8ab1f6d1cee5dd5e4eeb9e (refs/changes/64/5064/2)
Author: John Levon <john.levon@joyent.com>
Date:   2018-11-21T10:23:06+00:00 (11 months ago)
    
    OS-7379 kvm-cmd should use the strap compiler
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/Makefile.joyent b/Makefile.joyent
index 7987844..2c2c257 100644
--- a/Makefile.joyent
+++ b/Makefile.joyent
@@ -1,5 +1,5 @@
 #
-# Copyright 2015 Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 # Permission is hereby granted, free of charge, to any person obtaining a copy
 # of this software and associated documentation files (the "Software"), to deal
@@ -21,11 +21,12 @@
 #
 
 PROTO_AREA =	$(PWD)/../../../proto
+STRAP_AREA =	$(PWD)/../../../proto.strap
 KERNEL_SOURCE =	$(PWD)/../../illumos
 MDB_SOURCE =	$(KERNEL_SOURCE)/usr/src/cmd/mdb
 CTFBINDIR=	$(KERNEL_SOURCE)/usr/src/tools/proto/*/opt/onbld/bin/i386
 CSTYLE =	$(KERNEL_SOURCE)/usr/src/tools/scripts/cstyle
-CC =		$(PROTO_AREA)/usr/bin/gcc
+CC =		$(STRAP_AREA)/usr/bin/gcc
 INSTALL =	/usr/sbin/install
 
 QEMU_CPPFLAGS = \
@@ -92,16 +93,15 @@ DMOD_SRCS = \
 world: qemu qemu.so
 
 config-host.mak:
-	PATH=$(CTFBINDIR):$(PATH) CONFIGURE_ONLY=1 ./build.sh
+	STRAP_AREA=$(STRAP_AREA) PATH=$(CTFBINDIR):$(PATH) CONFIGURE_ONLY=1 ./build.sh
 
 qemu: config-host.mak
 	echo "Building world"
-	PATH=$(CTFBINDIR):$(PATH) V=1 gmake all
+	STRAP_AREA=$(STRAP_AREA) PATH=$(CTFBINDIR):$(PATH) V=1 gmake all
 
 qemu.so: $(DMOD_SRCS)
 	$(CC) $(DMOD_CPPFLAGS) $(DMOD_CFLAGS) $(DMOD_LDFLAGS) -o $@ \
 	    $(DMOD_SRCS) $(DMOD_LIBS)
-		
 
 update:
 	git pull --rebase
@@ -112,7 +112,7 @@ manifest:
 mancheck_conf:
 
 install: world
-	DESTDIR=$(DESTDIR) PATH=$(CTFBINDIR):$(PATH) V=1 gmake install
+	STRAP_AREA=$(STRAP_AREA) DESTDIR=$(DESTDIR) PATH=$(CTFBINDIR):$(PATH) V=1 gmake install
 	$(INSTALL) -f $(DESTDIR)/usr/lib/mdb/proc/amd64/ qemu.so
 
 clean:
diff --git a/build.sh b/build.sh
index 553a0ea..ab8a051 100755
--- a/build.sh
+++ b/build.sh
@@ -24,7 +24,7 @@ fi
 
 if [[ ! -e ${PNGLIB}/libpng.a ]]; then
     (cd ${PNGDIR} && \
-	CC="${CC:-${DESTDIR}/usr/bin/gcc}" \
+	CC="${CC:-${STRAP_AREA}/usr/bin/gcc}" \
 	LDFLAGS="-m64 -L${DESTDIR}/usr/lib/amd64 -L${DESTDIR}/lib/amd64" \
 	CPPFLAGS="-isystem ${DESTDIR}/usr/include" \
 	CFLAGS="-m64" ./configure --disable-shared && \
@@ -35,7 +35,7 @@ fi
 
 echo "==> Running configure"
 KVM_DIR="${KVM_DIR:-$(cd `pwd`/../kvm; pwd)}"
-CC="${CC:-${DESTDIR}/usr/bin/gcc}"
+CC="${CC:-${STRAP_AREA}/usr/bin/gcc}"
 XCFLAGS="-fno-builtin -I${PNGINC} -isystem ${DESTDIR}/usr/include -msave-args"
 XLDFLAGS="-nodefaultlibs -L${PNGLIB} -L${DESTDIR}/usr/lib/amd64 -L${DESTDIR}/lib/amd64"
 XLDFLAGS="${XLDFLAGS} -Wl,-zfatal-warnings -Wl,-zassert-deflib"
